components/kubernetes/kubernetes_object_ref.go

```diff
@@ -18,7 +18,7 @@ type KubernetesObjRefOutput struct {
 	LabelSelectors map[string]string `json:"labelSelectors"`
 }
 
-type KubernetesObjectRef struct {
+type KubernetesObjectRef struct { // nolint:revive, We want to keep the name as <Component>Output
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
What is revive complaining about here? The comment seems to be a copy paste from above that does not fit here

Reply:
I was getting the same error as the other types, complaining about the name starting with the name of the package. 
```
Error: components/kubernetes/kubernetes_object_ref.go:21:6: exported: type name will be used as kubernetes.KubernetesObjectRef by other packages, and that stutters; consider calling this ObjectRef (revive)
  type KubernetesObjectRef struct {
```
Should I rename it?

Reply:
no it's fine, I had it yesterday on `fakeintake`, thinking we should share the same package probably, to help discoverability, but that would be for another PR and not related to your changes. Thank you!